### ISSUE TITLE
zip.compress promise does not resolve if compressing 3 or more files

### ISSUE THREAD

**Issue by mbertogliati** [deno-library/compress#20]
This one is tricky, it took me a while to find the cause of it.
If I have an array of 3 files, and then compress them and wrap them in a Promise.all(), that promise does not resolve.
Minimal reproducible example:
```
import { zip } from "jsr:@deno-library/compress";
import { describe, it} from "jsr:@std/testing/bdd";

describe("zip", () => {
  it("should compress N files in a row", async () => {
    const N = 3;
    await Deno.mkdir("/tmp/zip-test", { recursive: true });
    const files: string[] = Array.from({ length: N }, (_, i) => `/tmp/zip-test/file_${i}.txt`);
    await Promise.all(files.map((file, i) => Deno.writeTextFile(file, `Hello, World! ${i}`)));
    await Promise.all(files.map((file, i) => zip.compress(file, `/tmp/zip-test/file_${i}.zip`)));
    await Deno.remove("/tmp/zip-test", { recursive: true });
  });
});

```
For any N >= 3 this test does not pass as the compress promise does not resolve. The compressed files do appear where they are supposed to, but the promise hangs.
Maybe I'm doing something wrong here, so I would be glad if someone can point me in the right direction.

**Comment by fuxingZhang** [comment]
The parameter of zip.compress is a file or a directory, and if you want to compress multiple independent files, you should use `jsr:@zip-js/zip-js`.
> https://jsr.io/@zip-js/zip-js

**Comment by gmzacharydovel** [comment]
This looks like if you call `zip.compress` more than 3 times the library freezes. I'm seeing this as well as I'm trying to create a zip file for each lambda entry point for AWS.

I'm trying the other library you posted @fuxingZhang , but this seems like a pretty big bug where the command must be storing something internally and not clearing it between command calls.

**Comment by fuxingZhang** [comment]
@gmzacharydovel Thank you for your feedback. 

This library is designed for compressing a folder or a single file. If you are looking to manually add specific files to a zip file, you might want to consider using a lower-level library like `jsr:@zip-js/zip-js`. 
> https://jsr.io/@zip-js/zip-js
This library is not posted by me. I just used it.

This library has 3.6k stars, which indicates it’s generally reliable, so any issues may stem from the way it’s being used. I recommend reviewing the documentation carefully. If you still encounter problems, please open an issue on the library's GitHub page and provide your code, so the developers can verify and assist with your issue.
> https://github.com/gildas-lormeau/zip.js

**Comment by gmzacharydovel** [comment]
I already moved on from this library a while ago because it's garbage to the often available zip cli command.  I needed something that works.

Yes, I did go over your documentation back in the day. Your documentation has nothing written about this library not working when trying to zip multiple files.  Maybe you haven't actually read it?  Maybe there should be something on your readme to save time of other users?

**Comment by mbertogliati** [comment]
Yeah, back in the day I just changed the library. This one is not reliable at all
